### ISSUE TITLE
Issues with v0.11.0b1

### ISSUE THREAD

**Issue by ErikBjare** [ActivityWatch/activitywatch#607]
I thought things were stable enough to be released, but apparently not.

I pulled the release, renamed it to v0.11.0b1, and marked it as a prerelease.

## Issues

 - [x] aw-qt SIGSEGVs when tray icon is clicked
 	- I really thought I had fixed this in https://github.com/ActivityWatch/activitywatch/pull/604...
 	- Fixed, see: https://github.com/ActivityWatch/activitywatch/issues/607#issuecomment-862187836
 - [x] aw-watcher-afk crashes due to missing hidden import (fixed in https://github.com/ActivityWatch/aw-watcher-afk/commit/4845e0f5d9be63b62fe99e22d5493f598293fbd0 and https://github.com/ActivityWatch/activitywatch/commit/d0c49c71768d9d59b85fc2ea9b8eee5ff151b6dd)
 - [x] Version not set from tag in CI (how did we do this in Travis CI?)
	 - Fixed in https://github.com/ActivityWatch/aw-server/commit/5f42ca64096d46291e091e62a189f8608c07c0da
 - [ ] Artifacts not automatically uploaded from GitHub Actions to the release
    - Might just do this manually for now.
 - More?

**Comment by ErikBjare** [comment]
Trying to fix the SIGSEGV. Did a diff between v0.10 (works fine0 and v0.11.0b1 (SIGSEGV when trayicon right-clicked), and got this:

```
diff activitywatch-v0.11.0b1/ldd-no-addr.out activitywatch-v0.10/ldd-no-addr.out
```

```diff
35a36
> 	libselinux.so.1 => ./libselinux.so.1 
38d38
< 	liblz4.so.1 => ./liblz4.so.1 
41,138d40
< libQt5EglFSDeviceIntegration.so.5:
< 	linux-vdso.so.1 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	libQt5DBus.so.5 => ./libQt5DBus.so.5 
< 	libfontconfig.so.1 => ./libfontconfig.so.1 
< 	libfreetype.so.6 => ./libfreetype.so.6 
< 	libEGL.so.1 => /usr/lib/libEGL.so.1 
< 	libmtdev.so.1 => ./libmtdev.so.1 
< 	libinput.so.10 => ./libinput.so.10 
< 	libxkbcommon.so.0 => ./libxkbcommon.so.0 
< 	libGL.so.1 => /usr/lib/libGL.so.1 
< 	libQt5Gui.so.5 => ./libQt5Gui.so.5 
< 	libudev.so.1 => ./libudev.so.1 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libdbus-1.so.3 => ./libdbus-1.so.3 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libexpat.so.1 => ./libexpat.so.1 
< 	libpng16.so.16 => ./libpng16.so.16 
< 	libz.so.1 => ./libz.so.1 
< 	libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 
< 	libevdev.so.2 => ./libevdev.so.2 
< 	librt.so.1 => /usr/lib/librt.so.1 
< 	libwacom.so.2 => ./libwacom.so.2 
< 	libGLX.so.0 => /usr/lib/libGLX.so.0 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libsystemd.so.0 => ./libsystemd.so.0 
< 	libgudev-1.0.so.0 => ./libgudev-1.0.so.0 
< 	libgobject-2.0.so.0 => ./libgobject-2.0.so.0 
< 	libX11.so.6 => ./libX11.so.6 
< 	liblzma.so.5 => ./liblzma.so.5 
< 	liblz4.so.1 => ./liblz4.so.1 
< 	libgcrypt.so.20 => ./libgcrypt.so.20 
< 	libffi.so.6 => ./libffi.so.6 
< 	libxcb.so.1 => /usr/lib/libxcb.so.1 
< 	libgpg-error.so.0 => ./libgpg-error.so.0 
< 	libXau.so.6 => ./libXau.so.6 
< 	libXdmcp.so.6 => ./libXdmcp.so.6 
< 	libbsd.so.0 => ./libbsd.so.0 
< libQt5EglFsKmsSupport.so.5:
< 	linux-vdso.so.1 
< 	libQt5EglFSDeviceIntegration.so.5 => ./libQt5EglFSDeviceIntegration.so.5 
< 	libEGL.so.1 => /usr/lib/libEGL.so.1 
< 	libQt5Gui.so.5 => ./libQt5Gui.so.5 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
< 	libdrm.so.2 => /usr/lib/libdrm.so.2 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	libQt5DBus.so.5 => ./libQt5DBus.so.5 
< 	libfontconfig.so.1 => ./libfontconfig.so.1 
< 	libfreetype.so.6 => ./libfreetype.so.6 
< 	libmtdev.so.1 => ./libmtdev.so.1 
< 	libinput.so.10 => ./libinput.so.10 
< 	libxkbcommon.so.0 => ./libxkbcommon.so.0 
< 	libGL.so.1 => /usr/lib/libGL.so.1 
< 	libudev.so.1 => ./libudev.so.1 
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 
< 	libz.so.1 => ./libz.so.1 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libdbus-1.so.3 => ./libdbus-1.so.3 
< 	libexpat.so.1 => ./libexpat.so.1 
< 	libpng16.so.16 => ./libpng16.so.16 
< 	libevdev.so.2 => ./libevdev.so.2 
< 	librt.so.1 => /usr/lib/librt.so.1 
< 	libwacom.so.2 => ./libwacom.so.2 
< 	libGLX.so.0 => /usr/lib/libGLX.so.0 
< 	libsystemd.so.0 => ./libsystemd.so.0 
< 	libgudev-1.0.so.0 => ./libgudev-1.0.so.0 
< 	libgobject-2.0.so.0 => ./libgobject-2.0.so.0 
< 	libX11.so.6 => ./libX11.so.6 
< 	liblzma.so.5 => ./liblzma.so.5 
< 	liblz4.so.1 => ./liblz4.so.1 
< 	libgcrypt.so.20 => ./libgcrypt.so.20 
< 	libffi.so.6 => ./libffi.so.6 
< 	libxcb.so.1 => /usr/lib/libxcb.so.1 
< 	libgpg-error.so.0 => ./libgpg-error.so.0 
< 	libXau.so.6 => ./libXau.so.6 
< 	libXdmcp.so.6 => ./libXdmcp.so.6 
< 	libbsd.so.0 => ./libbsd.so.0 
163,164d64
< 	libbsd.so.0 => ./libbsd.so.0 
< 	librt.so.1 => /usr/lib/librt.so.1 
170,220d69
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	libgssapi_krb5.so.2 => ./libgssapi_krb5.so.2 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libkrb5.so.3 => ./libkrb5.so.3 
< 	libk5crypto.so.3 => ./libk5crypto.so.3 
< 	libcom_err.so.2 => ./libcom_err.so.2 
< 	libkrb5support.so.0 => ./libkrb5support.so.0 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libkeyutils.so.1 => ./libkeyutils.so.1 
< 	libresolv.so.2 => /usr/lib/libresolv.so.2 
< libQt5Qml.so.5:
< 	linux-vdso.so.1 
< 	libQt5Network.so.5 => ./libQt5Network.so.5 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
< 	librt.so.1 => /usr/lib/librt.so.1 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	libz.so.1 => ./libz.so.1 
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	libgssapi_krb5.so.2 => ./libgssapi_krb5.so.2 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libkrb5.so.3 => ./libkrb5.so.3 
< 	libk5crypto.so.3 => ./libk5crypto.so.3 
< 	libcom_err.so.2 => ./libcom_err.so.2 
< 	libkrb5support.so.0 => ./libkrb5support.so.0 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libkeyutils.so.1 => ./libkeyutils.so.1 
< 	libresolv.so.2 => /usr/lib/libresolv.so.2 
< libQt5QmlModels.so.5:
< 	linux-vdso.so.1 
< 	libQt5Qml.so.5 => ./libQt5Qml.so.5 
< 	libQt5Network.so.5 => ./libQt5Network.so.5 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
225,228d73
< 	librt.so.1 => /usr/lib/librt.so.1 
< 	libz.so.1 => ./libz.so.1 
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	libgssapi_krb5.so.2 => ./libgssapi_krb5.so.2 
232,256d76
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libkrb5.so.3 => ./libkrb5.so.3 
< 	libk5crypto.so.3 => ./libk5crypto.so.3 
< 	libcom_err.so.2 => ./libcom_err.so.2 
< 	libkrb5support.so.0 => ./libkrb5support.so.0 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libkeyutils.so.1 => ./libkeyutils.so.1 
< 	libresolv.so.2 => /usr/lib/libresolv.so.2 
< libQt5Quick.so.5:
< 	linux-vdso.so.1 
< 	libQt5QmlModels.so.5 => ./libQt5QmlModels.so.5 
< 	libQt5Qml.so.5 => ./libQt5Qml.so.5 
< 	libQt5Gui.so.5 => ./libQt5Gui.so.5 
< 	libQt5Network.so.5 => ./libQt5Network.so.5 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
< 	libGL.so.1 => /usr/lib/libGL.so.1 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	librt.so.1 => /usr/lib/librt.so.1 
< 	libz.so.1 => ./libz.so.1 
258,261d77
< 	libgssapi_krb5.so.2 => ./libgssapi_krb5.so.2 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
265,270d80
< 	libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 
< 	libGLX.so.0 => /usr/lib/libGLX.so.0 
< 	libkrb5.so.3 => ./libkrb5.so.3 
< 	libk5crypto.so.3 => ./libk5crypto.so.3 
< 	libcom_err.so.2 => ./libcom_err.so.2 
< 	libkrb5support.so.0 => ./libkrb5support.so.0 
272,278d81
< 	libX11.so.6 => ./libX11.so.6 
< 	libkeyutils.so.1 => ./libkeyutils.so.1 
< 	libresolv.so.2 => /usr/lib/libresolv.so.2 
< 	libxcb.so.1 => /usr/lib/libxcb.so.1 
< 	libXau.so.6 => ./libXau.so.6 
< 	libXdmcp.so.6 => ./libXdmcp.so.6 
< 	libbsd.so.0 => ./libbsd.so.0 
305,373d107
< 	libbsd.so.0 => ./libbsd.so.0 
< 	librt.so.1 => /usr/lib/librt.so.1 
< libQt5WaylandClient.so.5:
< 	linux-vdso.so.1 
< 	libQt5Gui.so.5 => ./libQt5Gui.so.5 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
< 	libwayland-cursor.so.0 => ./libwayland-cursor.so.0 
< 	libwayland-client.so.0 => ./libwayland-client.so.0 
< 	libfontconfig.so.1 => ./libfontconfig.so.1 
< 	libfreetype.so.6 => ./libfreetype.so.6 
< 	libz.so.1 => ./libz.so.1 
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	libQt5DBus.so.5 => ./libQt5DBus.so.5 
< 	libGL.so.1 => /usr/lib/libGL.so.1 
< 	libxkbcommon.so.0 => ./libxkbcommon.so.0 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libffi.so.6 => ./libffi.so.6 
< 	librt.so.1 => /usr/lib/librt.so.1 
< 	libexpat.so.1 => ./libexpat.so.1 
< 	libpng16.so.16 => ./libpng16.so.16 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libdbus-1.so.3 => ./libdbus-1.so.3 
< 	libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 
< 	libGLX.so.0 => /usr/lib/libGLX.so.0 
< 	libsystemd.so.0 => ./libsystemd.so.0 
< 	libX11.so.6 => ./libX11.so.6 
< 	liblzma.so.5 => ./liblzma.so.5 
< 	liblz4.so.1 => ./liblz4.so.1 
< 	libgcrypt.so.20 => ./libgcrypt.so.20 
< 	libxcb.so.1 => /usr/lib/libxcb.so.1 
< 	libgpg-error.so.0 => ./libgpg-error.so.0 
< 	libXau.so.6 => ./libXau.so.6 
< 	libXdmcp.so.6 => ./libXdmcp.so.6 
< 	libbsd.so.0 => ./libbsd.so.0 
< libQt5WebSockets.so.5:
< 	linux-vdso.so.1 
< 	libQt5Network.so.5 => ./libQt5Network.so.5 
< 	libQt5Core.so.5 => ./libQt5Core.so.5 
< 	libpthread.so.0 => /usr/lib/libpthread.so.0 
< 	libstdc++.so.6 => ./libstdc++.so.6 
< 	libm.so.6 => /usr/lib/libm.so.6 
< 	libgcc_s.so.1 => ./libgcc_s.so.1 
< 	libc.so.6 => /usr/lib/libc.so.6 
< 	libz.so.1 => ./libz.so.1 
< 	libdl.so.2 => /usr/lib/libdl.so.2 
< 	libgssapi_krb5.so.2 => ./libgssapi_krb5.so.2 
< 	libicui18n.so.56 => ./libicui18n.so.56 
< 	libicuuc.so.56 => ./libicuuc.so.56 
< 	libicudata.so.56 => ./libicudata.so.56 
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
< 	/usr/lib64/ld-linux-x86-64.so.2 
< 	libkrb5.so.3 => ./libkrb5.so.3 
< 	libk5crypto.so.3 => ./libk5crypto.so.3 
< 	libcom_err.so.2 => ./libcom_err.so.2 
< 	libkrb5support.so.0 => ./libkrb5support.so.0 
< 	libpcre.so.3 => ./libpcre.so.3 
< 	libkeyutils.so.1 => ./libkeyutils.so.1 
< 	libresolv.so.2 => /usr/lib/libresolv.so.2 
399,400d132
< 	libbsd.so.0 => ./libbsd.so.0 
< 	librt.so.1 => /usr/lib/librt.so.1 
402a135,136
> 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
> 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
405d138
< 	libz.so.1 => ./libz.so.1 
412,426d144
< 	libxcb-icccm.so.4 => ./libxcb-icccm.so.4 
< 	libxcb-image.so.0 => ./libxcb-image.so.0 
< 	libxcb-shm.so.0 => ./libxcb-shm.so.0 
< 	libxcb-util.so.1 => ./libxcb-util.so.1 
< 	libxcb-keysyms.so.1 => ./libxcb-keysyms.so.1 
< 	libxcb-randr.so.0 => ./libxcb-randr.so.0 
< 	libxcb-render-util.so.0 => ./libxcb-render-util.so.0 
< 	libxcb-render.so.0 => ./libxcb-render.so.0 
< 	libxcb-shape.so.0 => ./libxcb-shape.so.0 
< 	libxcb-sync.so.1 => ./libxcb-sync.so.1 
< 	libxcb-xfixes.so.0 => ./libxcb-xfixes.so.0 
< 	libxcb-xinerama.so.0 => ./libxcb-xinerama.so.0 
< 	libxcb-xkb.so.1 => ./libxcb-xkb.so.1 
< 	libxcb.so.1 => /usr/lib/libxcb.so.1 
< 	libXext.so.6 => ./libXext.so.6 
428,429c146,148
< 	libxkbcommon-x11.so.0 => ./libxkbcommon-x11.so.0 
< 	libxkbcommon.so.0 => ./libxkbcommon.so.0 
---
> 	libXi.so.6 => ./libXi.so.6 
> 	libXrender.so.1 => ./libXrender.so.1 
> 	libxcb.so.1 => /usr/lib/libxcb.so.1 
431,432d149
< 	libgthread-2.0.so.0 => ./libgthread-2.0.so.0 
< 	libglib-2.0.so.0 => ./libglib-2.0.so.0 
436a154
> 	libpcre.so.3 => ./libpcre.so.3 
438c156,157
< 	libpng16.so.16 => ./libpng16.so.16 
---
> 	libz.so.1 => ./libz.so.1 
> 	libpng12.so.0 => ./libpng12.so.0 
445a165
> 	libXext.so.6 => ./libXext.so.6 
448d167
< 	libpcre.so.3 => ./libpcre.so.3 
450c169
< 	libbsd.so.0 => ./libbsd.so.0 
---
> 	libselinux.so.1 => ./libselinux.so.1 
453d171
< 	liblz4.so.1 => ./liblz4.so.1 

```


Noticed the huge diff, and that a lot more libraries than needed seem to be included in v0.11.0b1 (unless there's good reason for some). Notably, libfontconfig and libfreetype were missing in v0.10, and then I found this: https://gitlab.com/inkscape/inkscape/-/issues/491

Turns out, libfontconfig and libfreetype shouldn't be included. If I rm these libs (or use LD_PRELOAD to override), it works fine.

**Comment by ErikBjare** [comment]
Issues fixed, v0.11.0 has been released!
